tests/utils/profilehandler/data_preparation.go

```diff
@@ -38,28 +44,111 @@ func PreparePrefix(profilePrefix string, nameLength int) string {
+// if zones are empty list it will list the zones and pick according to multi-zone parameter. when multi-zone=true, 3 zones will be pickup
+func PrepareSubnets(vpcClient *vpc_client.VPC, region string, zones []string, multiZone bool) (map[string][]string, error) {
+	resultMap := map[string][]string{}
+	if len(zones) == 0 {
+		log.Logger.Info("Got no zones indicated. List the zones and pick from the listed zones")
+		resultZones, err := vpcClient.AWSClient.ListAvaliableZonesForRegion(region, "availability-zone")
+		if err != nil {
+			return resultMap, err
+		}
+		zones = resultZones[0:1]
+		if multiZone {
+			zones = resultZones[0:3]
+		}
+	}
+	for _, zone := range zones {
+		subnetMap, err := vpcClient.PreparePairSubnetByZone(zone)
+		if err != nil {
+			return resultMap, err
+		}
+		for subnetType, subnet := range subnetMap {
+			if _, ok := resultMap[subnetType]; !ok {
+				resultMap[subnetType] = []string{}
+			}
+			resultMap[subnetType] = append(resultMap[subnetType], subnet.ID)
+		}
 	}
+
+	return resultMap, nil
 }
 
-func PrepareKMSKeyDummy(region string) string {
-	return ""
+func PrepareProxy(vpcClient *vpc_client.VPC, zone string, sshPemFile string, caFile string) (*ProxyDetail, error) {
+	_, privateIP, caContent, err := vpcClient.LaunchProxyInstance("", zone, sshPemFile)
+	if err != nil {
+		return nil, err
+	}
+	_, err = common.CreateFileWithContent(caFile, caContent)
+	if err != nil {
+		return nil, err
+	}
+	return &ProxyDetail{
+		HTTPsProxy:       fmt.Sprintf("https://%s:8080", privateIP),
+		HTTPProxy:        fmt.Sprintf("http://%s:8080", privateIP),
+		CABundleFilePath: caFile,
+		NoProxy:          "quay.io",
+	}, nil
+}
+
+func PrepareKMSKey(region string, multiRegion bool, testClient string, hcp bool) (string, error) {
+	keyArn, err := kms_key.CreateOCMTestKMSKey(region, multiRegion, testClient)
+	if err != nil {
+		return keyArn, err
+	}
+	if hcp {
+		kms_key.AddTagToKMS(keyArn, region, map[string]string{
+			"red-hat": "true",
+		})
+	}
+	return keyArn, err
+}
+
+func ElaborateKMSKeyForSTSCluster(client *rosacli.Client, cluster string, etcdKMS bool) error {
+	jsonData, err := client.Cluster.GetJSONClusterDescription(cluster)
+	if err != nil {
+		return err
+	}
+	accountRoles := []string{
+		jsonData.DigString("aws", "sts", "role_arn"),
+	}
+	operaorRoleMap := map[string]string{}
+	keyArn := jsonData.DigString("aws", "kms_key_arn")
+	if etcdKMS {
+		keyArn = jsonData.DigString("aws", "etcd_encryption", "kms_key_arn")
+	}
+	operatorRoles := jsonData.DigObject("aws", "sts", "operator_iam_roles").([]interface{})
+	for _, operatorRole := range operatorRoles {
+		role := operatorRole.(map[string]interface{})
+		operaorRoleMap[role["name"].(string)] = role["role_arn"].(string)
+	}
+	region := jsonData.DigString("region", "id")
+	isHCP := jsonData.DigBool("hypershift", "enabled")
+	err = kms_key.ConfigKMSKeyPolicyForSTS(keyArn, region, isHCP, accountRoles, operaorRoleMap)
+	return err
```

Comment:
`return kms_key.ConfigKMSKeyPolicyForSTS(keyArn, region, isHCP, accountRoles, operaorRoleMap` could be more beautiful.

Reply:
Added log output to make it resolved